### PR TITLE
ticket #9  Recipe/User edit routes

### PR DESCRIPTION
**Pull request by abin-thomas**
To add, edit and delete a recipe as well as to edit a user profile.

### CODE REVIEW

File: server/models/views.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    request, make_response, 
+)
+from models.recipe import Recipe
+from models.user import User
+from models.profile import Profile
+
+
+@app.route("/recipe/add/", methods=["POST"])
+def add_recipe(chef_id):
+    if request.method == "POST":
```
Comment (sina-jamshidi) [inline server/models/views.py]: you don't need this check when there's already only one method allowed on the route

File: server/models/views.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        request_dict = request.form.to_dict()
+        recipe_info = {}
+        for field in Recipe.get_fields():
+            if request_dict.get(field, None) != None:
+                if field =='price':
+                    recipe_info[field]= request_dict.get(field)/100
```
Comment (sina-jamshidi) [inline server/models/views.py]: We're storing prices as integers so division shouldn't be necessary

File: server/models/views.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            return response
+        except AssertionError as e:
+            # Validation problem
+            return {
+                'status': 401,
+                'message': "%s" % (e)
+            }
+        
+
+@app.route("/recipe/edit/<int:recipe_id>", methods=["POST"])
```
Comment (sina-jamshidi) [inline server/models/views.py]: Best practice wise, the route should be `recipe/<id>/edit`, and the method should be `PUT`.

File: server/models/views.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                'status': 401,
+                'message': "%s" % (e)
+            }
+
+@app.route("/recipe/delete/<int:recipe_id>", methods=["GET"])
+def delete_recipe(recipe_id):
+    try:
+        Recipe.delete(recipe_id)
+        return{
+            'status': 201,
```
Comment (sina-jamshidi) [inline server/models/views.py]: this should be status 200. See [HTTP status codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status) and check your other statuses too.

File: server/models/views.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            response_dict['image_urls'] = recipe.get_image_url_list
+            response_dict['ingredients'] = recipe.get_ingredients_list
+            response_dict['required_items'] = recipe.get_utensil_list
+            
+            response = make_response(response_dict,201)
+            return response
+        except AssertionError as e:
+            # Validation problem
+            return {
+                'status': 401,
```
Comment (sina-jamshidi) [inline server/models/views.py]: status 401 means `unauthorized`. So it doesn't apply here. 400 might be better

File: server/models/views.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+            'message': "Recipe updated successfully."
+        }
+        except AssertionError as e:
+            # Validation problem
+            return {
+                'status': 401,
+                'message': "%s" % (e)
+            }
+
+@app.route("/recipe/delete/<int:recipe_id>", methods=["GET"])
```
Comment (sina-jamshidi) [inline server/models/views.py]: you should use a `"DELETE"` method here. See [HTTP methods](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods)

File: server/models/views.py (hunk 1 above)
Comment (abin-thomas) [reply server/models/views.py]: Got it. Done

File: server/models/views.py (hunk 5 above)
Comment (abin-thomas) [reply server/models/views.py]: Changed to 400.

File: server/models/views.py (hunk 4 above)
Comment (abin-thomas) [reply server/models/views.py]: Done.

**Pull request merged**